openpathsampling/utils/storage_handlers.py

```diff
@@ -0,0 +1,108 @@
+from abc import ABC, abstractmethod
+import os
+import pathlib
+import shutil
+
+import logging
+_logger = logging.getLogger(__name__)
+
+
+
+class StorageHandler(ABC):
+    """Abstract treatment of a key-value-like file/object store.
+
+    This is generally assuming file-based semantics. This may typically mean
+    putting things into a temporary directory. This is particularly focused
+    on checkpointing, where we will copy the data to put it in a zip file.
+    """
+    @abstractmethod
+    def store(self, storage_label, source_path):
+        raise NotImplementedError()
+
+    @abstractmethod
+    def load(self, storage_label, target_path):
+        raise NotImplementedError()
+
+    @abstractmethod
+    def delete(self, storage_label):
+        raise NotImplementedError()
+
+    @abstractmethod
+    def __contains__(self, storage_label):
+        raise NotImplementedError()
+
+    def transfer(self, storage_label, source_path):
+        """Transfer a file to the storage label from the source path.
+
+        In some cases, this can be made faster than store followed by
+        os.remove, so this method can be overridden. (Example: moving ona
+        file system is faster than copying.)
+        """
+        self.store(storage_label, source_path)
+        os.remove(source_path)
+
+    @abstractmethod
+    def list_directory(self, storage_label):
+        raise NotImplementedError()
+
+
+class LocalFileStorageHandler(StorageHandler):
+    """Concrete implementation of StorageHandler for local files.
+    """
+    def __init__(self, root):
+        self.root = pathlib.Path(root)
+        self.root.mkdir(parents=True, exist_ok=True)
+
+    def store(self, storage_label, source_path):
+        local_path = self.root / storage_label
+        local_path.parent.mkdir(parents=True, exist_ok=True)
+        shutil.copyfile(source_path, self.root / storage_label)
+
+    def load(self, storage_label, target_path):
+        target_path.parent.mkdir(parents=True, exist_ok=True)
+        local_path = self.root / storage_label
+        _logger.debug("Copying file from {str(local_path)} "
+                      f"to {str(target_path)}")
+        shutil.copyfile(local_path, target_path)
```

Comment:
this could use docs, specifically of the difference between load and store (load seems to do something I would not expect from the name)